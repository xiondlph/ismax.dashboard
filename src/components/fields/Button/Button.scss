$dashboard-field-button-bg: $dashboard-header-user-photo-color;
$dashboard-field-button-disabled-color: #ececec;

.dashboard-field-button {
  background-color: $dashboard-field-button-bg;
  color: $dashboard-text-color-white;
  padding: 5px 10px;
  cursor: pointer;
  outline: none;
  border: none;

  &:disabled {
    background-color: lighten($dashboard-field-button-bg, 10%);
    color: $dashboard-field-button-disabled-color;
    cursor: default;
  }

  &:hover:not(:disabled) {
    background-color: adjust-hue($dashboard-field-button-bg, 10%);
  }

  &:active:hover {
    background-color: darken($dashboard-field-button-bg, 10%);
  }
}