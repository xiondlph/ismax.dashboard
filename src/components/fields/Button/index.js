/**
 * Компонент кнопки формы
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import './Button.scss'

export default class Button extends Component {
    render() {
        const { type, disabled } = this.props

        return (
            <button
                className={classNames({
                    'dashboard-field-button': true
                })}
                type={type}
                disabled={disabled}
            >
                { this.props.children }
            </button>
        )
    }
}

Button.propTypes = {
    type: PropTypes.string.isRequired,
    disabled: PropTypes.bool
}