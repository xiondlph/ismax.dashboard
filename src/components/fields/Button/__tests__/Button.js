import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../index'

describe('Button', () => {
    test('Проверка Button снапшота', () => {
        const
            component = renderer.create(
                <Button type='button' disabled={false}>Action</Button>
            )

        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('Проверка disable состояния для button', () => {
        const
            component = renderer.create(
                <Button type='button' disabled={true}>Action</Button>
            )

        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})