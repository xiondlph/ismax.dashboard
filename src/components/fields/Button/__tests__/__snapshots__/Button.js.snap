// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Проверка Button снапшота 1`] = `
<button
  className="dashboard-field-button"
  disabled={false}
  type="button"
>
  Action
</button>
`;

exports[`Button Проверка disable состояния для button 1`] = `
<button
  className="dashboard-field-button"
  disabled={true}
  type="button"
>
  Action
</button>
`;
